lib/vmdb/plugins/registry/bundler.rb

```diff
@@ -0,0 +1,65 @@
+module Vmdb
+  class Plugins
+    module Registry
+      class Bundler
+        class Plugin
+          def initialize(spec)
+            @spec = spec
+          end
+
+          def name
+            @spec.name
+          end
+
+          def root
+            @root ||= Pathname.new @spec.gem_dir
+          end
+        end
+
+        include Singleton
+
+        # FIXME:  Don't rely on this constant for determining non-provider
+        # based plugins
+        #
+        # Instead find a way to do the equaivalent of `vmdb_plugin?` without
+        # having to load the class (file in the root, modularize the code
+        # containing `def vmdb_root?`, etc.)
+        NON_PROVIDER_PLUGINS = %w(automation_engine content ui-classic).freeze
+
+        attr_reader :plugins, :provider_plugins
+
+        def initialize
+          @plugins = fetch_plugins
+        end
+
+        def provider_name(plugin)
+          plugin.name.gsub("manageiq-providers-", "")
+        end
+
+        def provider_plugin?(plugin)
+          provider_plugins.include? plugin
+        end
+
+        private
+
+        def fetch_plugins
+          plugin_gems.map do |spec|
+            Plugin.new(spec)
+          end
+        end
+
+        def plugin_gems
+          ::Bundler.locked_gems.specs.select do |spec|
```

Comment:
@jrafanie Yes, and this is also only used in `fetch_plugins`.

Sidenote:  I originally appended the `map {|spec| Plugin.new` to the end of this block, and had all of it inside of the `fetch_plugins` method, but "rubocop didn't like that" (_he says in a whiney sarcastic voice_).

The reason I mention that this was originally in `fetch_plugins` is because `fetch_plugins` is also the only place that `Rails.application` is ever accessed as well in it's respective class.  The only difference is that the `engine` interface exposes `root` and `name` directly, which is what we end up using to build the plugin map, but a "proxy interface", aka `Vmdb::Plugins::Registry::Bundler::Plugin`, was needed to allow that same interface to exist for the bundler one.